### DISCUSSION TITLE
%LocalAppData% unset on Windows when using Actions for CI

### DISCUSSION THREAD

**Discussion by mvdan** [community/community#25019] [Actions]
<p>While testing a Go project on Windows, I came across an error:</p>
<pre><code>build cache is required, but could not be located: GOCACHE is not defined and %LocalAppData% is not defined
</code></pre>
<p>I can set GOCACHE or LocalAppData, but that seems unnecessary. LocalAppData should be set on Windows builds.</p>
<p>This will make testing most Go tools on Windows hard, as they will usually require writing to GOCACHE, which falls back to a directory under LocalAppData when unset.</p>
<p>Full logs: <a href="https://github.com/mvdan/unparam/runs/219056273" rel="nofollow noopener">https://github.com/mvdan/unparam/runs/219056273</a></p>

**Comment by chrispat** [comment, marked as answer]
<p>I am not sure what is going on there.  I forked your repo and dumped the environment variables before go was run and LOCALAPPDATA is there <a href="https://github.com/chrispat/unparam/commit/fc77b63dbde48b7ba1fdd5fc29e476f59875b94b/checks?check_suite_id=281593816#step:4:58" rel="nofollow noopener">https://github.com/chrispat/unparam/commit/fc77b63dbde48b7ba1fdd5fc29e476f59875b94b/checks?check_suite_id=281593816#step:4:58</a>. </p>
<p>I don’t know a lot about Go so I am not sure where to start debugging.  Is it possbile your test-script model that runs your go test scripts is not passing the entire environment along when it launches go?</p>

**Comment by mvdan** [comment]
<p>You’re absolutely right, this was an oversight on my part. I had <a href="https://github.community/t/temp-is-broken-on-windows/16137">another issue</a> with environment variables on Windows, so I just assumed this was an issue on your side too :slight_smile:</p>
